subsys/bluetooth/shell/bt.c

```diff
@@ -193,65 +193,84 @@ static bool data_cb(struct bt_data *data, void *user_data)
 	}
 }
 
-static void print_data_set(uint8_t type, uint8_t set_value_len,
-			   const uint8_t *scan_data, uint8_t scan_data_len)
+static void print_data_hex(const uint8_t *data, uint8_t len)
 {
-	uint8_t min_value_len = MIN(set_value_len, scan_data_len);
+	shell_fprintf(ctx_shell, SHELL_INFO, "0x");
+	for (int8_t i = len-1; i >= 0; i--) {
+		shell_fprintf(ctx_shell, SHELL_INFO, "%02x", data[i]);
+	}
+}
 
-	shell_fprintf(ctx_shell, SHELL_INFO, "%*sType 0x%02x: ",
-		      strlen(scan_response_label), "",
-		      type);
-	for (uint8_t i = 0U; i < (scan_data_len / min_value_len); i++) {
-		if (i > 0L) {
-			shell_fprintf(ctx_shell, SHELL_INFO, ", ");
-		}
+static void print_data_set(uint8_t set_value_len, bool single_value,
+			   const uint8_t *scan_data, uint8_t scan_data_len)
+{
+	uint8_t idx = 0;
 
-		shell_fprintf(ctx_shell, SHELL_INFO, "0x");
-		for (uint8_t j = 0U; j < min_value_len; j++) {
-			shell_fprintf(ctx_shell, SHELL_INFO, "%02x",
-				      scan_data[i * min_value_len + j]);
+	do {
+		if (idx > 0) {
+			shell_fprintf(ctx_shell, SHELL_INFO, ",");
 		}
-	}
-
-	shell_fprintf(ctx_shell, SHELL_INFO, "\n");
+		print_data_hex(&scan_data[idx], set_value_len);
+		idx += set_value_len;
+		if (single_value)
+			break;
+	} while (idx+set_value_len <= scan_data_len);
 }
 
 static bool data_verbose_cb(struct bt_data *data, void *user_data)
 {
+	shell_fprintf(ctx_shell, SHELL_INFO, "%*sType 0x%02x: ",
+		strlen(scan_response_label), "", data->type);
+
 	switch (data->type) {
 	case BT_DATA_UUID16_SOME:
 	case BT_DATA_UUID16_ALL:
 	case BT_DATA_SOLICIT16:
+		print_data_set(2, false, data->data, data->data_len);
+		break;
 	case BT_DATA_SVC_DATA16:
-		print_data_set(data->type, 2, data->data, data->data_len);
+		print_data_set(2, true, data->data, data->data_len);
+		shell_fprintf(ctx_shell, SHELL_INFO, ",");
+		print_data_set(1, false, data->data+2, data->data_len-2);
```

Comment:
```suggestion
		print_data_set(1, false, data->data + 2, data->data_len - 2);
```
General comment: In Zephyr we use spaces around operators :)

Reply:
my local checkpatch didn't complain :) .. but will fix that